invenio_records_resources/services/files/service.py

```diff
@@ -65,13 +65,27 @@ def _get_record(self, id_, identity, action, file_key=None):
 
         return record
 
+    def _ensure_record_not_deleted(self, record, identity, include_deleted):
+        """Ensure that the record exists (not deleted) or raise."""
+        if record.is_draft:
```

Comment:
I am not sure about this part here, we are in records-resouces, we shouldn't know about drafts' existence in this module

Reply:
I changed it to be `record.is_published`